### ISSUE TITLE
Zoltan2: MultiJagged causing Epetra-based Panzer MiniEM RefMaxwell test to fail

### ISSUE THREAD

**Issue by brian-kelley** [trilinos/Trilinos#6424]
## Bug Report
@trilinos/zoltan2 
@trilinos/panzer 

### Description
In a fresh build from develop with tests and examples enabled, Panzer's ``PanzerMiniEM_MiniEM-BlockPrec_RefMaxwell_Epetra_MPI_4`` is failing. This is in panzer/mini-em/example/BlockPrec. It's really an example (not a normal test), but this is probably worth looking into. Changing the MueLu repartition algorithm in ``solverMueLuRefMaxwellEpetra.xml`` from multijagged to rcb makes it pass.

### Steps to Reproduce
1. SHA1:  8565fc6 (current develop as of 12/10 afternoon)
2. [Configure script](https://gist.github.com/brian-kelley/91addd5352420460bd365f034efd9077)
3. Error output I get:
```
******* WARNING *******
Level::Set: unable to store "AP reuse data" generated by factory RAPFactory[15](0x69009f0) on level 1, as it has not been requested and no keep flags were set for it

******* WARNING *******
Level::Set: unable to store "RAP reuse data" generated by factory RAPFactory[15](0x69009f0) on level 1, as it has not been requested and no keep flags were set for it
RefMaxwell (2,2): Build (MueLu::RepartitionFactory)
RefMaxwell (2,2): Build (MueLu::Zoltan2Interface)
Zoltan2 parameters:
----------
algorithm = multijagged
partitioning_approach = partition   [unused]
mj_premigration_option = 1   [unused]
num_global_parts = 3   [unused]
----------
RefMaxwell (2,2): Zoltan2 multijagged
  Teko: "buildInverse" could not construct the inverse operator using "Thyra::MueLuRefMaxwellPreconditionerFactory"
  
  *** THROWN EXCEPTION ***
  getIDsKokkosView called but not on Tpetra!
  ************************
terminate called after throwing an instance of 'terminate called after throwing an instance of 'std::exception'
std::exception'
  what():  std::exception  what():  std::exception

[s1042556:101025] *** Process received signal ***
[s1042556:101026] *** Process received signal ***
  Teko: "buildInverse" could not construct the inverse operator using "Teko::AutoClone<mini_em::FullMaxwellPreconditionerFactory, mini_em::FullMaxwellPreconditionerFactory>"
  
  *** THROWN EXCEPTION ***
  std::exception
  ************************
[s1042556:101026] Signal: Aborted (6)
```

**Comment by MicheldeMessieres** [comment]
I plan to resolve this tomorrow.

**Comment by MicheldeMessieres** [comment]
@brian-kelley After discussion it seems some other open issues with miultijagged should be higher priority so I did not address this yet but will soon.

**Comment by MicheldeMessieres** [comment]
@brian-kelley Do you by chance have a config which builds this example on white? I was having some issues with STK and boost I need to resolve. @kddevin If white is not good, maybe I could get access to an appropriate machine so we can share an exact config. I ask about white because I have access currently. Thanks.

**Comment by MicheldeMessieres** [comment]
@brian-kelley Never mind on the request. I think I have white building as I need it. Using the default build with your config above, which I think should properly generate the example since cuda and openmp are now off. Thanks.

**Comment by brian-kelley** [comment]
@MicheldeMessieres Here's a minimal configure that enables it, with the usual flags to speed up the build. On white you also need to help Trilinos find the BLAS TPL.
```
cmake \
 -DTPL_ENABLE_MPI=ON \
 -DCMAKE_BUILD_TYPE=Release \
 -DTrilinos_ENABLE_TESTS=ON \
 -DTrilinos_ENABLE_EXAMPLES=ON \
 -DTrilinos_ENABLE_Tpetra:BOOL=ON \
 -DTrilinos_ENABLE_Epetra:BOOL=ON \
 -DTrilinos_ENABLE_Panzer:BOOL=ON \
 -DTpetra_INST_INT_INT=ON \
 -DTPL_ENABLE_Matio=OFF \
 -DTrilinos_ENABLE_EXPLICIT_INSTANTIATION=ON \
 -DBUILD_SHARED_LIBS=ON \
../Trilinos
```

**Comment by MicheldeMessieres** [comment]
@brian-kelley Could you please clarify what you sourced before calling the config? I had tried just source /ascldap/users/medemes/trilinos/cmake/std/atdm/load-env.sh gnu.
But then needed to specify boost lib and ended up with stk errors. I understand we want a build with CUDA and OpenMP off for the example. I'm not sure if there is a build on white I can reference.

**Comment by alanw0** [comment]
As an aside, the next stk update into trilinos will remove the need for BoostLib. (Will still need some header-only boost stuff for a while.)

**Comment by brian-kelley** [comment]
@MicheldeMessieres I was using almost the same set of modules, but with some a couple different MPI and TPL versions. Today I did ``load-env.sh gnu`` and this exact configuration worked:

```
cmake \
  -DTPL_ENABLE_MPI=ON \
  -DCMAKE_BUILD_TYPE=Release \
  -DTrilinos_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_EXAMPLES=ON \
  -DTrilinos_ENABLE_Tpetra:BOOL=ON \
  -DTrilinos_ENABLE_Epetra:BOOL=ON \
  -DTrilinos_ENABLE_Panzer:BOOL=ON \
  -DTpetra_INST_INT_INT=ON \
  -DTPL_ENABLE_Matio=OFF \
  -DTrilinos_ENABLE_EXPLICIT_INSTANTIATION=ON \
  -DBUILD_SHARED_LIBS=ON \
  -DTPL_BLAS_LIBRARIES=/usr/lib64/libblas.so.3 \
  -DTPL_LAPACK_LIBRARIES=/usr/lib64/liblapack.so.3 \
  -DNetcdf_LIBRARY_DIRS=$NETCDF_ROOT/lib \
  -DBoostLib_LIBRARY_DIRS=$BOOST_ROOT/lib \
  ../Trilinos
```

**Comment by MicheldeMessieres** [comment]
@brian-kelley I tried a fresh login to white, completely clean build, and did load-env.sh gnu and then ran that config. But I get a build error:
`*** No rule to make target `/usr/lib64/liblapack.so.3', needed by `packages/teuchos/numerics/src/libteuchosnumerics.so.12.17'.  Stop.`
I'm on the latest develop. So just double checking there was no other action I take apart from the load_env.sh and config? Thanks.

**Comment by brian-kelley** [comment]
@MicheldeMessieres Argh, I'm so sorry, the system blas/lapack aren't available on the compute nodes (I've been running it on a login node). The netlib module defines BLAS_ROOT and LAPACK_ROOT. I tested this config on a rhel7T node and also the login node. As a side note, I didn't realize how much slower the login node is at building.
```
cmake \
  -DTPL_ENABLE_MPI=ON \
  -DCMAKE_BUILD_TYPE=Release \
  -DTrilinos_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_EXAMPLES=ON \
  -DTrilinos_ENABLE_Tpetra:BOOL=ON \
  -DTrilinos_ENABLE_Epetra:BOOL=ON \
  -DTrilinos_ENABLE_Panzer:BOOL=ON \
  -DTpetra_INST_INT_INT=ON \
  -DTPL_ENABLE_Matio=OFF \
  -DTrilinos_ENABLE_EXPLICIT_INSTANTIATION=ON \
  -DBUILD_SHARED_LIBS=ON \
  -DBLAS_LIBRARY_DIRS=$BLAS_ROOT/lib \
  -DLAPACK_LIBRARY_DIRS=$LAPACK_ROOT/lib \
  -DNetcdf_INCLUDE_DIRS=$NETCDF_ROOT/include \
  -DNetcdf_LIBRARY_DIRS=$NETCDF_ROOT/lib \
  -DBoost_INCLUDE_DIRS=$BOOST_ROOT/include \
  -DBoostLib_INCLUDE_DIRS=$BOOST_ROOT/include \
  -DBoostLib_LIBRARY_DIRS=$BOOST_ROOT/lib \
  ../Trilinos
```

**Comment by MicheldeMessieres** [comment]
@brian-kelley Got further but still having issues. I'm trying to login to rhel7T first.

I'm failing now with:
```
[ 10%] Linking CXX executable Epetra_BlockMap_test.exe
../../src/libepetra.so.12.17: undefined reference to `dggsvd_'
../../src/libepetra.so.12.17: undefined reference to `sggsvd_'
```

This is my config for Lapack:
```
Processing enabled TPL: LAPACK (enabled by TeuchosNumerics, disable with -DTPL_ENABLE_LAPACK=OFF)
-- LAPACK_LIBRARY_NAMES='lapack lapack_win32'
-- Searching for libs in LAPACK_LIBRARY_DIRS='/ascldap/users/projects/ppc64le-pwr8/netlib/3.8.0/gcc/7.2.0/lib'
-- Searching for a lib in the set "lapack lapack_win32":
--   Searching for lib 'lapack' ...
--     Found lib '/ascldap/users/projects/ppc64le-pwr8/netlib/3.8.0/gcc/7.2.0/lib/liblapack.a'
-- TPL_LAPACK_LIBRARIES='/ascldap/users/projects/ppc64le-pwr8/netlib/3.8.0/gcc/7.2.0/lib/liblapack.a'
```

My procedure is:
Fresh login to white
bsub -Is -n 16 -q rhel7T bash
source /ascldap/users/medemes/trilinos/cmake/std/atdm/load-env.sh gnu
Then run above config in a fresh empty build folder.
make NP=16

**Comment by brian-kelley** [comment]
@MicheldeMessieres I think deleting CMakeCache.txt will fix that error, since it may have not been updated when you switched to the right LAPACK.

**Comment by MicheldeMessieres** [comment]
@brian-kelley I was building a completely clean build folder (so not make clean) but mkdir newFolder.

**Comment by brian-kelley** [comment]
@MicheldeMessieres I found a workaround for the linker error, thanks to @jewatkins. I did a fresh login, did load-env.sh gnu, did the interactive bsub to rhel7T like you did above, and started with a fresh build, and this config works (I promise):

```
rm -rf CMakeFiles
rm -f CMakeCache.txt
cmake \
  -DTPL_ENABLE_MPI=ON \
  -DCMAKE_BUILD_TYPE=Release \
  -DTrilinos_ENABLE_TESTS=ON \
  -DTrilinos_ENABLE_EXAMPLES=ON \
  -DTrilinos_ENABLE_Tpetra:BOOL=ON \
  -DTrilinos_ENABLE_Epetra:BOOL=ON \
  -DTrilinos_ENABLE_Panzer:BOOL=ON \
  -DCMAKE_C_COMPILER=mpicc \
  -DCMAKE_CXX_COMPILER=mpicxx \
  -DTrilinos_ENABLE_Fortran=OFF \
  -DTpetra_INST_INT_INT=ON \
  -DTPL_ENABLE_Matio=OFF \
  -DTrilinos_ENABLE_EXPLICIT_INSTANTIATION=ON \
  -DBUILD_SHARED_LIBS=ON \
  -DBLAS_LIBRARY_DIRS=$BLAS_ROOT/lib \
  -DLAPACK_LIBRARY_DIRS=$LAPACK_ROOT/lib \
  -DNetcdf_INCLUDE_DIRS=$NETCDF_ROOT/include \
  -DNetcdf_LIBRARY_DIRS=$NETCDF_ROOT/lib \
  -DBoost_INCLUDE_DIRS=$BOOST_ROOT/include \
  -DBoostLib_INCLUDE_DIRS=$BOOST_ROOT/include \
  -DBoostLib_LIBRARY_DIRS=$BOOST_ROOT/lib \
  -DHAVE_dggsvd3_POST=1 \
  ../Trilinos
```

Also, I ran "make" from packages/panzer/mini-em/example/BlockPrec so as to build the minimum stuff (in particular, no tests from other packages).

**Comment by MicheldeMessieres** [comment]
@brian-kelley Thanks. That works. I'll have a PR to fix this soon.

**Comment by MicheldeMessieres** [comment]
@brian-kelley I just merged #6567 which will fix this.

**Comment by brian-kelley** [comment]
@MicheldeMessieres Great, thanks!
